File: sdk-php/InnovacionX.php
<?php

class InnovacionX {

    private $token;
    private $baseUrl;

    public function __construct($token, $baseUrl = "https://tu-dominio.com/api/public") {
        if (!$token) {
            throw new Exception("Falta x-public-token");
        }
        $this->token = $token;
        $this->baseUrl = $baseUrl;
    }

    private function request($endpoint, $method = "GET", $data = null) {
        $url = $this->baseUrl . $endpoint;

        $headers = [
            "Content-Type: application/json",
            "x-public-token: {$this->token}"
        ];

        $options = [
            "http" => [
                "header"  => implode("\r\n", $headers),
                "method"  => $method,
                "content" => $data ? json_encode($data) : null
            ]
        ];

        $context  = stream_context_create($options);
        $result = file_get_contents($url, false, $context);

        if ($result === FALSE) {
            throw new Exception("Error en la solicitud HTTP");
        }

        return json_decode($result, true);
    }

    public function getProductos() {
        return $this->request("/productos");
    }
}

File: sdk-python/innovacionx.py
import requests

class InnovacionX:
    def __init__(self, token, base_url="https://tu-dominio.com/api/public"):
        if not token:
            raise Exception("Falta x-public-token")
        self.token = token
        self.base_url = base_url

    def request(self, endpoint, method="GET", data=None):
        url = f"{self.base_url}{endpoint}"
        headers = {
            "x-public-token": self.token,
            "Content-Type": "application/json"
        }

        response = requests.request(method, url, json=data, headers=headers)

        if not response.ok:
            try:
                error = response.json()
                raise Exception(error.get("error", f"HTTP {response.status_code}"))
            except:
                raise Exception(f"HTTP {response.status_code}")

        return response.json()

    # ============================
    # 📦 Productos públicos
    # ============================
    def get_productos(self):
        return self.request("/productos")
